code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+/obj/effect/temp_visual/tendril_blood
+	name = "tendril blood"
+	icon_state = "tendril_blood"
+	duration = 0.5 SECONDS
+	layer = TANK_BARREL_LAYER
+
+/obj/effect/temp_visual/tendril_blood/Initialize()
+	. = ..()
+	animate(src, alpha = 65, time = 0.4)
+
+/obj/effect/ebeam/slime_tendril
+	name = "slimy tendril"
+	alpha = 200
+	layer = ABOVE_MOB_LAYER
+
+// ***************************************
+// *********** Acidic Trail
+// ***************************************
+/datum/action/xeno_action/acidic_trail
+	name = "Acidic Trail"
+	ability_name = "Acidic Trail"
+	action_icon_state = "acidic_trail"
+	desc = "Left click toggles a trail of acidic puddles when moving. Right click toggles an acid that will imbue your slashes with Intoxication."
+	cooldown_timer = 0.5 SECONDS
+	plasma_cost = 0
+	use_state_flags = XACT_USE_LYING|XACT_USE_BUCKLED|XACT_USE_STAGGERED|XACT_USE_NOTTURF|XACT_USE_BUSY|XACT_USE_CLOSEDTURF
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_SLIME_ACIDIC_TRAIL,
+		KEYBINDING_ALTERNATE = COMSIG_XENOABILITY_SLIME_ACIDIC_TRAIL_ALTERNATE,
+	)
+	/// Whether the trail of acidic puddles when moving is active or not.
+	var/acidic_steps = TRUE
+	/// Whether the acid that imbues your slashes with Intoxication is active or not.
+	var/acidic_slashes = TRUE
+
+/datum/action/xeno_action/acidic_trail/give_action(mob/living/L)
+	. = ..()
+	if(acidic_steps)
+		RegisterSignal(L, COMSIG_MOVABLE_MOVED, PROC_REF(acid_steps))
+	if(acidic_slashes)
+		RegisterSignal(L, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(acid_slashes))
+
+/datum/action/xeno_action/acidic_trail/remove_action(mob/living/L)
+	. = ..()
+	UnregisterSignal(L, COMSIG_MOVABLE_MOVED, COMSIG_XENOMORPH_ATTACK_LIVING)
+
+/datum/action/xeno_action/acidic_trail/action_activate()
+	if(acidic_steps)
+		UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+		owner.balloon_alert(owner, "Acidic Steps OFF")
+		acidic_steps = FALSE
+		return
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(acid_steps))
+	owner.balloon_alert(owner, "Acidic Steps ON")
+	acidic_steps = TRUE
+
+/datum/action/xeno_action/acidic_trail/proc/acid_steps(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/xeno_owner = mover
+	new /obj/effect/xenomorph/spray/slime(get_turf(xeno_owner), SLIME_MOVEMENT_ACID_DURATION, SLIME_MOVEMENT_ACID_DAMAGE)
+	for(var/obj/O in get_turf(xeno_owner))
+		O.acid_spray_act(xeno_owner)
```

Comment:
Why only objects, is this intentional?
Also sidenote a more clear varname than O would be nice
Also note to self todo acidsprayact should be in spray code?

Reply:
It was mostly to give it something of an edge against objects. They don't matter as much as living targets do; you could easily spam shuffling movement on a target to delete their health, so I decided against it.

I can change the varname though, that's done.

Reply:
On second thought, it doesn't need this. Removed.